app/assets/javascripts/discourse/app/components/sidebar/user/sections.js

```diff
@@ -1,21 +1,21 @@
 import Component from "@glimmer/component";
-import { customSections as sidebarCustomSections } from "discourse/lib/sidebar/custom-sections";
 import { getOwner, setOwner } from "@ember/application";
 import { inject as service } from "@ember/service";
-import { cached } from "@glimmer/tracking";
 
 export default class SidebarUserSections extends Component {
   @service siteSettings;
   @service currentUser;
   @service site;
 
-  @cached
-  get customSections() {
-    return sidebarCustomSections.map((customSection) => {
-      const section = new customSection({ sidebar: this });
-      setOwner(section, getOwner(this));
-      return section;
-    });
+  constructor() {
+    super(...arguments);
+
+    this.customSections =
+      this.args.panel?.sections?.map((customSection) => {
+        const section = new customSection({ sidebar: this });
```

Comment:
Personally, I'm not a big fan of passing `this` in because this couples the component and the class together. Is there a way we can pass in on the properties we need here?